### DISCUSSION TITLE
Templating with multiple revisions

### DISCUSSION THREAD

**Discussion by stouset** [jj-vcs/jj#3075] [Q&A] [answered]
Commands like `jj log` apply the template to each revision independently. When templating multiple revisions, I might want to do something like "join each revision with a comma and space". The closest I've come is something like this:

```
jj log --ignore-working-copy --no-graph --revisions @-::@ --template 'change_id.shortest() ++ ", "'
rk, os, 
```

but that results in an extra separator at the end.

Is there a way to apply a template function like `join()` *between revisions*?

**Comment by yuja** [comment, marked as answer]
> Is there a way to apply a template function like `join()` _between revisions_?

No, there isn't right now.

Mercurial has `revset()` template function, and it could be used conceptually as `log -r@ -T 'revset("$1-::$1", commit_id).map(|c| c.change_id().shortest()).join(", ")'`. Maybe jj can have something like that.
